### PR TITLE
gzip static assets

### LINKED ISSUE #5443: Feature Request: Enable HTTP Compression for JS/CSS files
Current HTTP traffic does not respond with HTTP compression. Using HTTP Compression on HTML/JS type traffic would reduce the amount of web traffic, decrease the initial screen loading time and provider a faster user experience. As an example the initial login gets 2.8 MB of resources. The majority of that traffic is JS content which could be compressed.

We can check if the browser supports compression by examining the request headers and returned a compressed file using standard golang library. 

(here is an example: https://github.com/kubernetes/dashboard/blob/master/src/app/backend/handler/gziphandler.go)

### PR DESCRIPTION
**Pull request by ibeckermayer**
Adds middlewear to gzip static assets, ignoring common compressed image formats (png/jpg)

**Before:**
Screen Shot 2021-02-11 at 9 34 35 AM

**After:**
Screen Shot 2021-02-11 at 9 33 29 AM

closes https://github.com/gravitational/teleport/issues/5443

### CODE REVIEW

File: lib/web/gziphandler.go
Comment (alex-kovoy) [inline lib/web/gziphandler.go:44]:
```golang
return strings.HasSuffix(r.URL.Path, ".png") || strings.HasSuffix(r.URL.Path, ".jpeg") 
```

File: lib/web/gziphandler.go
Comment (alex-kovoy) [inline lib/web/gziphandler.go:21]: could you remind what is this one needed? For example, we have headers which explicitly tell the browsers not to do the content sniffing as the browsers might be tricked to do things they are not supposed to.

Comment (ibeckermayer) [reply lib/web/gziphandler.go:22]: resolved in slack: [content sniffing is skipped](https://github.com/golang/go/issues/31753) if content-encoding header is set, so we need sniff it ourselves before writing the compressed data

**Review by alex-kovoy — APPROVED** [review]
Lets wait for another approval (2 in total for Teleport repo) before merge.

File: lib/web/gziphandler.go
Comment (russjones) [inline lib/web/gziphandler.go:38]: Instead of creating a new writer for every request can you reuse the writer? You can just cut/paste the code from below.

```
// writerPool is a sync.Pool for shared gzip writers.
// each gzip writer allocates a lot of memory
// so it makes sense to reset the writer and reuse the
// internal buffers to avoid too many objects on the heap
```

https://github.com/gravitational/teleport/blob/master/lib/events/sessionlog.go#L615-L633

File: lib/web/gziphandler.go
Comment (russjones) [inline lib/web/gziphandler.go:27]: Does not need to be exported, not used outside this package.

**Comment by alex-kovoy** [comment]
@russjones ^^^

File: lib/web/gziphandler.go
Comment (russjones) [inline lib/web/gziphandler.go:1]: Missing copyright header.

**Review by russjones — APPROVED** [review]
Code looks fine, please add test coverage and you're good to go.

**Comment by alex-kovoy** [comment]
@ibeckermayer @russjones i think it's OK to skip the test for this one as essentially we are testing 4 lines of code.

**Pull request merged**
